server/controllers/index.js

```diff
@@ -8,75 +8,81 @@ let currentUsers = [];
 
 /* Check if player is logged in */
 router.get('/auth', (req, res) => {
-    if (req.session.user) {
+    if (req.session.user) { /* The user session exists, authorised */
         res.status(200).json({ cookie: req.session.user });
-    } else {
-        res.sendStatus(401);
+    } else { /* The user session does not exist, unauthorised */
+        res.status(401).json({ message: "You need to log in with a username to enter the game."});
     }
 });
 
 /* Log in the user */
 router.post('/auth/login', (req, res) => {
     const user = currentUsers.find((user) => user.username === req.body.username);
-    if (!user) {
+    if (!user) { /* User does not already exist */
         req.session.user = req.body.username;            
         const user = {
             username: req.body.username
         };        
-        if (gameLogic.canJoinGame(user.username)) {
+        if (gameLogic.canJoinGame(user.username)) { /* Game has not been started yet */
             currentUsers.push(user);
             gameLogic.joinGame(user, () => {
                 res.sendStatus(200);
             });
-        } else {
-            res.status(400).json({message: "Game has already started"});
+        } else { /* Game has already began */
+            res.status(400).json({message: "Game has already started."});
         }
-    } else {
-        res.sendStatus(409);
+    } else { /* Username is taken, conflict error */
+        res.status(409).json({message: "Username already exists."});
     }    
 });
 
 /* Log out the user */
-router.post('/auth/logout', (req, res) => {
-    const user = currentUsers.find((user) => user.username === req.session.user);
-    if (user) {
+router.get('/auth/logout', auth, (req, res) => {
+    if (gameLogic.quitGame(user)) { /* Game has finished, or has not been started yet */
         currentUsers = currentUsers.filter((otherUser) => otherUser !== user);
-        if (gameLogic.quitGame(user)) {
-            req.session.destroy();
-            gameLogic.removePlayer(user.username);
-            res.sendStatus(200);
-        } else {
-            res.sendStatus(400)
-        }
-    } else {
-        res.status(404).json({message: "Cannot log out: user does not exist."});
+        req.session.destroy();
+        gameLogic.removePlayer(user.username);
+        res.sendStatus(200);
+    } else { /* Game has started, method not allowed */
+        res.status(400).json({message: "Cannot log out of a running game."});
```

Comment:
general point around this file / api - if the gameLogic.quitGame() or any others throw an error, are they handled gracefully in the gameLogic method? If not it might be worth using try catch blocks because one error will crash your server for everyone

Reply:
Yeah, Ronan suggested using try and catch instead as well, I just pushed a new commit